asserts/snapasserts/snapasserts.go

```diff
@@ -192,12 +195,24 @@ func DeriveSideInfo(snapPath string, model *asserts.Model, db Finder) (*snap.Sid
 // on the device.
 func DeriveSideInfoFromDigestAndSize(snapPath string, snapSHA3_384 string, snapSize uint64, model *asserts.Model, db Finder) (*snap.SideInfo, error) {
 	// get relevant assertions and reconstruct metadata
-	a, err := db.Find(asserts.SnapRevisionType, map[string]string{
+	headers := map[string]string{
 		"snap-sha3-384": snapSHA3_384,
-	})
-	if err != nil {
+	}
+	a, err := db.Find(asserts.SnapRevisionType, headers)
+	if err != nil && !asserts.IsNotFound(err) {
 		return nil, err
 	}
+	if a == nil {
+		// non-default provenance?
+		cands, err := db.FindMany(asserts.SnapRevisionType, headers)
```

Comment:
AFAICT, both `Find` and `FindMany` will look for the assertion with a default value for the "provenance" primary key, right? I'd expect that with the same headers, the result would be the same but clearly that's not true. Why is that?

Reply:
no, where you saw this? FindMany doesn't add the defaults, as left out values mean anything is accepted for them, we would not be able to find things by only partial part of the primary key, which is exactly what we are trying to do here.

This backstore test I think shows this:
https://github.com/snapcore/snapd/blob/master/asserts/fsbackstore_test.go#L653

with the behavor you described we would only get the first result "k1/o1-defl"

Does this make sense?

Reply:
Nonetheless it's maybe worth to write an extra test that reaches the error len(cands) > 1

Reply:
I pushed such a test

Reply:
I see, I misread that in the code. That makes more sense, thanks